src/dynamicTypes/npmDependencies.js

```diff
@@ -1,22 +1,68 @@
-const getPackageRawDeclaration = name =>
+const getDefaultPackageRawDeclaration = name =>
   `
     declare module '${name}';
     declare module '${name}/*';
   `;
 
-function getPackagesTypeDeclarations(dependencies = {}) {
+const BASE_PACKAGE_TYPES_URL = `https://manage.wix.com/_serverless/package-types-resolver`;
```

Comment:
> what's the contract of this method? what does it do?

`https://manage.wix.com/_serverless/package-types-resolver/<package>@<version>`

will 301 redirect to whatever is written in the `pacakge.json`'s `types`field, for example:

`https://manage.wix.com/_serverless/package-types-resolver/@wix/react-velo`
---301-->
`https://static.parastorage.com/unpkg/@wix/react-velo@1.0.19/types/index.d.ts`

Why is this even required?
To do 1 request instead of 2 (read `package.json`'s types field and then request actual types file, also see 2nd point below.)

> will it work for any npm?
<img src=https://user-images.githubusercontent.com/326402/158379184-8c211dbc-47a7-4c3d-a630-1df1178d1558.png width="200" height="112" />

There are multiple aspects to this:
1. I've restricted it to only `@wix` scope in the [server side](https://github.com/wix-private/package-types-resolver/blob/83532a1c161ef747500e168be1182779f4283f96/serverless/package-types-resolver/src/serverless.ts#L11-L13), as I don't want to allow loading types for any package on the internet without a strategy to how we ensure those types are safe.

2. The `types` field might contain a file which has references to multiple other `.d.ts`s, there are two solutions to this:
        
    2a. We ask developers who wish their types to work on Velo, bundle their types with something like [npm-dts](https://github.com/vytenisu/npm-dts) (by @vytenisu ) or their own solution.
        
    2b. We create the bundle ourselves from the `.d.ts`, and make  `package-types-resolver` serve a redirect to that bundled `.d.ts` instead what's written in the `types` filed in `package.json` as it currently does.
        
 Personally I prefer (a) 🙃 , at least for the start.

Reply:
After discussing with @dlvhdr and @nemni8 we've deiced to extend the service above:
```
https://manage.wix.com/_serverless/package-types-resolver/<package>@<version>
```

To:
1. Resolve the types field
2. Follow the `d.ts` file in `types` and bundle it
3. Cache and return the bundled types result.

Reply:
Just to follow up:
1-2 - I've created https://github.com/yurynix/remote-type-bundler
3. I've refactored the endpoint above to:
```
curl -v https://manage.wix.com/_serverless/package-types-resolver/@wix/react-velo@1.0.20\?bundle\=true\&wrapWithModuleDeclare\=true
```

Reply:
amaze few questions
1. why on your github and not wix? when we merge this we'll take ownership for this server
2. which/what cache are u adding?

Reply:
1. I'm ready to move to whatever repo you'll tell me 🙃 
2. Wix Media Platform, check it out:
```
curl -v https://manage.wix.com/_serverless/package-types-resolver/@wix/react-velo@1.0.20\?bundle\=true\&wrapWithModuleDeclare\=true
```

```
https://wixmp-ee582f73d5044f7b3bb01ffb.wixmp.com/@wix/react-velo@1.0.20.wrapped.bundle.d.ts
```